### PR TITLE
Adding random instanceDir

### PR DESCRIPTION
**Pull request by mohammedzee1000**
Signed-off-by: Mohammed Zeeshan Ahmed <mohammed.zee1000@gmail.com>

### CODE REVIEW

File: pkg/worker/worker.go [hunk 1]
```diff
@@ -410,8 +410,9 @@ func (w *Worker) tearDownTests(oldsuccess bool, ex executor.Executor, workDir st
 func (w *Worker) test(nd *node.Node) (bool, error) {
 	var err error
 	var ex executor.Executor
-	workDir := strings.ReplaceAll(w.cimsg.RcvIdent, ".", "_")
-	repoDir := filepath.Join(workDir, w.repoDir)
+	baseWorkDir := util.GetBaseWorkDir(w.cimsg.RcvIdent)
+	instanceWorkDir := filepath.Join(baseWorkDir, util.GetInstanceWorkdirName())
```
Comment (prietyc123) [inline pkg/worker/worker.go:414]: ok, after each trigger are we going to delete this temp directory? I mean in a single pr just say job fails at first, it has created some dir. So we expect it to be deleted just after build fails. Then for the next trigger in the same pr, we will have some other dir, no?

Comment (mohammedzee1000) [reply pkg/worker/worker.go:414]: Ok, basically there is going to be a base workdir and an instance dir.

Base workdir is common for the target being built (eg odo-pr-build-pr-1) and will always be the target of cleanup, both before a new run and after a successful run. Cleanup will not happen for failed run.

The actual build will happen in instance dir which will be a random string whose path will be baseworkdir/instanceworkdir. Eg repo will be cloned in baseworkdir/instancewormdir/repo which is also where your scripts will execute.

At the end you will get a unique directory for every run which is still traceable and ideally you should not have more than once instance dir at a time

Comment (prietyc123) [reply pkg/worker/worker.go:414]: > Cleanup will not happen for failed run.

Then how you are expecting clean up for such cases.

Comment (mohammedzee1000) [reply pkg/worker/worker.go:414]: We don't we keep the env intact for failed run so ppl can investigate if they want.

We can have separate cleanup for cases where prsbare closed etc leaving no chance of cleanup the normal way

Comment (prietyc123) [reply pkg/worker/worker.go:414]: > We don't we keep the env intact for failed run so ppl can investigate if they want.
> 
> We can have separate cleanup for cases where prsbare closed etc leaving no chance of cleanup the normal way

Ok, so how about keeping that directory into machine for two days. We can have cronjob scheduled into our VM to delete such files after 2 days of their creation.

Comment (mohammedzee1000) [reply pkg/worker/worker.go:414]: Yes and we should do do this directly on vm. It should not be done by ci-firewall.

We can find the age of directory using commands quite easily.so have scripts that iterates over dirs and if deletes them if older than

Comment (prietyc123) [reply pkg/worker/worker.go:414]: /lgtm

**Pull request merged**